realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -92,6 +92,7 @@ private SyncConfiguration(File directory,
                                 RealmProxyMediator schemaMediator,
                                 RxObservableFactory rxFactory,
                                 Realm.Transaction initialDataTransaction,
+                                boolean readOnly,
```

Comment:
Yes, it is way too long, but it is only called from one place (inside the Builder). Only other alternative would be having a no-arg constructor and a bunch of setter methods, but that looks even worse?